public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -140,6 +141,8 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
   }
 
   private _activationHandler() {
+    this.runIsEmptyCheck();
```

Comment:
@ivanortegaalba i've added a behavior that tracks if body is empty (i love this solution so thank you for the suggestion 🚀 ). I've found an edge case when an empty dashboard is saved (not sure if this ever happens), and in this case empty dashboard modal won't show up because there's no way to know if a dashboard is empty unless running `runIsEmptyCheck`. LMWYT, because an alternative would be to just set isEmpty to false, but that wouldn't cover this edge case.